SharkCage/CageManager/CageLabeler.h

```diff
@@ -208,5 +325,185 @@ bool GetBottomFromMonitor(int &monitor_bottom)
 
 	monitor_bottom = monitor_info.rcMonitor.bottom;
 
+	return true;
+}
+
+static bool ShowExitButton(HWND &hwnd)
+{
+	int bottom;
+	if (!GetBottomFromMonitor(bottom))
+	{
+		std::wcout << L"Failed to get bottom of monitor" << std::endl;
+		return false;
+	}
+
+	gotodesk_button = ::CreateWindowEx(
+		NULL,
+		L"BUTTON",
+		L"Exit",
+		WS_VISIBLE | WS_CHILD | BS_DEFPUSHBUTTON,
+		10,
+		bottom - 44,
+		cage_width - 23,
+		34,
+		hwnd,
+		NULL,
+		NULL,
+		NULL);
+
+
+	if (gotodesk_button != NULL)
+	{
+		::SendMessage(gotodesk_button, BM_SETIMAGE, NULL, NULL);
+	}
+	else
+	{
+		std::wcout << L"Failed to create logout button Err " << ::GetLastError() << std::endl;
+		return false;
+	}
+
+	return true;
+}
+
+static bool ShowConfigMetadata(HWND &hwnd)
+{
+	app_name_title = ::CreateWindowEx(
+		NULL,
+		TEXT("STATIC"),
+		app_name.c_str(),
+		SS_LEFT | WS_VISIBLE | WS_CHILD,
+		10,
+		cage_width + 115,
+		150,
+		34,
+		hwnd,
+		NULL,
+		NULL,
+		NULL);
+
+	app_name_restart_button = ::CreateWindowEx(
+		NULL,
+		L"BUTTON",
+		L"Restart",
+		WS_VISIBLE | WS_CHILD | BS_DEFPUSHBUTTON,
+		cage_width - 112,
+		cage_width + 106,
+		100,
+		34,
+		hwnd,
+		NULL,
+		NULL,
+		NULL);
+
+	if (additional_app_name.has_value())
+	{
+		additional_app_app_title = ::CreateWindowEx(
+			NULL,
+			TEXT("STATIC"),
+			additional_app_name.value().c_str(),
+			SS_LEFT | WS_VISIBLE | WS_CHILD,
+			10,
+			cage_width + 155,
+			150,
+			34,
+			hwnd,
+			NULL,
+			NULL,
+			NULL);
+
+		additional_app_restart_button = ::CreateWindowEx(
+			NULL,
+			L"BUTTON",
+			L"Restart",
+			WS_VISIBLE | WS_CHILD | BS_DEFPUSHBUTTON,
+			cage_width - 112,
+			cage_width + 148,
+			100,
+			34,
+			hwnd,
+			NULL,
+			NULL,
+			NULL);
+	}
+
+	app_hash_text_title = ::CreateWindowEx(
+		NULL,
+		TEXT("STATIC"),
+		L"App SHA-512 (truncated):",
+		SS_LEFT | WS_VISIBLE | WS_CHILD,
+		10,
+		cage_width + 250,
+		cage_width - 20,
+		34,
+		hwnd,
+		NULL,
+		NULL,
+		NULL);
+
+	app_hash_text = ::CreateWindowEx(
+		NULL,
+		TEXT("STATIC"),
+		app_hash.c_str(),
+		SS_LEFT | WS_VISIBLE | WS_CHILD,
+		10,
+		cage_width + 275,
+		cage_width - 30,
+		34,
+		hwnd,
+		NULL,
+		NULL,
+		NULL);
+
+	if (app_name_title != NULL && app_name_restart_button != NULL)
+	{
+		HFONT default_font = ::CreateFont(
```

Comment:
According to the [doc](https://msdn.microsoft.com/en-us/library/windows/desktop/dd183499%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396) created fonts should be deleted with `::DeleteObject` when they are no longer needed, i.e. after the control using them is destroyed.

Just wanted to mention it, but I think it is okay to ignore this as the process finishes shortly after these controls are destroyed, anyway...